gobblin-utility/src/main/java/org/apache/gobblin/util/FileListUtils.java

```diff
@@ -125,8 +125,17 @@ public static List<FileStatus> listFilesRecursively(FileSystem fs, Path path, Pa
   public static List<FileStatus> listFilesRecursively(FileSystem fs, Path path, PathFilter fileFilter,
       boolean applyFilterToDirectories)
       throws IOException {
-    return listFilesRecursivelyHelper(fs, Lists.newArrayList(), fs.getFileStatus(path), fileFilter,
-        applyFilterToDirectories, false);
+    return listFilesRecursively(fs, fs.getFileStatus(path), fileFilter, applyFilterToDirectories);
+  }
+
+  /**
+   * Helper method to list out all files under a specified {@link FileStatus}. If applyFilterToDirectories is false, the supplied
+   * {@link PathFilter} will only be applied to files.
+   */
+  public static List<FileStatus> listFilesRecursively(FileSystem fs, FileStatus beneathThisFile, PathFilter fileFilter,
```

Comment:
Why do we need this change?

Reply:
just avoids accessing the `getFileStatus` twice (since we already had been doing this internally for the recursion helper)

Reply:
in some measure it brings minor symmetry, since the method both takes and returns `FileStatus`